src/hooks/useArweave.ts

```diff
@@ -0,0 +1,70 @@
+import Arweave from 'arweave';
+import { decrypt } from 'ecies-geth';
+import { utils } from 'ethers';
+import { useNetworkConfig } from 'lib/config';
+import { useCallback } from 'react';
+
+export function useArweave() {
+  const { arweaveConfig } = useNetworkConfig();
+  const arweave = arweaveConfig.host ? Arweave.init(arweaveConfig) : undefined;
+
+  const arweaveNotReadyMsg = 'Arweave instance not ready!';
+
+  const fetchArweaveFileFallback = useCallback(
+    async (arweaveTxId: string): Promise<any> => {
+      if (!arweave) {
+        console.error(arweaveNotReadyMsg);
+        return;
+      }
+
+      try {
+        const res = await arweave.api.get(`/${arweaveTxId}`, {
+          responseType: 'arraybuffer',
+        });
+        return res.data;
+      } catch (error) {
+        throw new Error(`Error fetching arweave file: ${error}`);
+      }
+    },
+    [arweave]
+  );
+
+  const fetchArweaveFile = useCallback(
+    async (arweaveTxId: string): Promise<Uint8Array | undefined> => {
+      if (!arweave) {
+        console.error(arweaveNotReadyMsg);
+        return;
```

Comment:
It's extremely unlikely that `arweave` would be undefined at this (and all such similar) point. But, you never know.